### DISCUSSION TITLE
Use of JSI

### DISCUSSION THREAD

**Discussion by charles-goode** [invertase/react-native-firebase#5785] [Ideas] [answered]
Does this library use or have plans to use JSI in order to speed up communication between the native SDKs and the Javascript layer? Seems like a no brainer for a huge performance gain.

**Comment by mikehardy** [comment, marked as answer]
It would be nice, sure, the serialization in firestore especially could probably be faster.
There are no plans currently as - even if slower than JSI - it works and we've got admob and mlkit to finish extracting plus some full API misses from the underlying SDKs (firestore bundles, RTDB get etc), but I'm always open to collaborate with people on PRs!

**Reply by Nantris** [reply]
Any change to this conclusion 2 years later? Or still no plans? I know there's already a lot on the plates for the developers of this library so I understand the need to be picky about what's implemented.
